[PL-50647] Update SMP external self-managed TimescaleDB with pg_cron extension details (#6553)
harness/developer-hub: docs/self-managed-enterprise-edition/advanced-configurations/external-db/use-an-external-sm-timescaledb.md

4. Upgrade Harness with your updated `override.yaml` file.
-->

## Enable the pg_cron extension

> To utilize the power of scheduled tasks within your PostgreSQL database, you must enable the `pg_cron` extension on your database instance.

Author: ```suggestion
To utilize scheduled tasks within your PostgreSQL database, you must enable the `pg_cron` extension on your database instance.
```

`pg_cron` functions as a cron-based job scheduler that operates internally within the database as an extension. It adheres to the familiar syntax of regular cron while empowering you to schedule PostgreSQL commands directly from within your database environment.

This approach offers more efficiency compared to relying solely on TimescaleDB. With the capabilities of `pg_cron`, the system efficiently schedules and executes aggregation tasks essential for generating metric data.

> ### Installation of pg_cron on Postgres database

Author: ```suggestion
### Install pg_cron on your Postgres database
```

For installations on Red Hat, CentOS, Fedora, or Amazon Linux running PostgreSQL 16 using the PostgreSQL Global Development Group (PGDG):

```bash
# Install the pg_cron extension
sudo yum install -y pg_cron_16
```

For installations on Debian or Ubuntu with PostgreSQL 16 using `apt.postgresql.org`:

```bash
# Install the pg_cron extension
sudo apt-get -y install postgresql-16-cron
```

Alternatively, you can opt to build `pg_cron` from its source:

```bash
git clone https://github.com/citusdata/pg_cron.git
cd pg_cron
```

Ensure that `pg_config` is within your path, for instance:

```bash
export PATH=/usr/pgsql-16/bin:$PATH
make && sudo PATH=$PATH make install
```

> To ensure the `pg_cron` background worker starts with PostgreSQL, add `pg_cron` to `shared_preload_libraries` within the `postgresql.conf` file:

Reviewer: ```suggestion
To ensure that the `pg_cron` background worker starts with PostgreSQL, add `pg_cron` to `shared_preload_libraries` within the `postgresql.conf` file:
```

> To ensure the `pg_cron` background worker starts alongside PostgreSQL, add `pg_cron` to `shared_preload_libraries` within the `postgresql.conf` file:

Author: ```suggestion
To ensure the `pg_cron` background worker starts with PostgreSQL, add `pg_cron` to `shared_preload_libraries` within the `postgresql.conf` file:
```

```conf
# add to postgresql.conf
# required to load pg_cron background worker on start-up
shared_preload_libraries = 'pg_cron'
```

> By default, the `pg_cron` background worker assumes its metadata tables exist within the Postgres database. However, you can customize this behavior by setting the `cron.database_name` configuration parameter in the `postgresql.conf` file:

Reviewer: ```suggestion
By default, the `pg_cron` background worker assumes that its metadata tables exist within the Postgres database. However, you can customize this behavior by setting the `cron.database_name` configuration parameter in the `postgresql.conf` file:
```

> By default, the `pg_cron` background worker assumes its metadata tables exist within the "postgres" database. However, you can customize this behavior by setting the `cron.database_name` configuration parameter in the `postgresql.conf` file:

Author: ```suggestion
By default, the `pg_cron` background worker assumes its metadata tables exist within the Postgres database. However, you can customize this behavior by setting the `cron.database_name` configuration parameter in the `postgresql.conf` file:
```

```conf
# add to postgresql.conf
# optionally, specify the database in which the pg_cron background worker should run (defaults to postgres)
cron.database_name = 'postgres'
```

> Previously restricted to GMT time, `pg_cron` now enables you to adapt timezones by setting `cron.timezone` in the `postgresql.conf` file. For example:

Author: ```suggestion
Previously restricted to GMT, `pg_cron` now enables you to adapt timezones by setting `cron.timezone` in the `postgresql.conf` file. For example:
```

```conf
# add to postgresql.conf

> # optionally, specify the timezone in which the pg_cron background worker should run (defaults to GMT). E.g:

Author: ```suggestion
# optionally, specify the timezone in which the pg_cron background worker should run (defaults to GMT). For example:
```

cron.timezone = 'PRC'
```

> After restarting PostgreSQL, execute `CREATE EXTENSION pg_cron;` as a superuser to create the necessary pg_cron functions and metadata tables:

Author: ```suggestion
After restarting PostgreSQL, execute `CREATE EXTENSION pg_cron;` as a superuser to create the necessary pg_cron functions and metadata tables:
```
